src/components/file/files-s3-bucket.factory.ts

```diff
@@ -7,7 +7,12 @@ export const FilesBucketToken = Symbol('FilesBucket');
 export const FilesBucketFactory = {
   provide: FilesBucketToken,
   useFactory: (s3: S3, config: ConfigService) => {
-    return new S3Bucket(s3, config.files.bucket, {});
+    this.s3 = new S3({ apiVersion: '2006-03-01' });
+    return new S3Bucket(s3, config.files.bucket, {
+      accessKeyId: config.aws.accessKeyId,
+      secretAccessKey: config.aws.accessKeyId,
+      region: config.aws.region,
+    });
```

Comment:
S3 configures automatically just like the `aws-cli` does. There's no need to explicitly pull in env vars.

Reply:
ya, to my surprise its working.